src/HandleVisit.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { Hotspot3D } from "./DataStructures";
+
+export type VisitedState = Record<string, Record<string, boolean>>;
```

Comment:
Changing this type to the following will let you use `?.` without an ESLint warning:
```ts
export type VisitedState = Partial<Record<string, Record<string, boolean>>>;
```

The `Partial` makes it so that the result of accessing an element will have a `Record<string, boolean> | undefined` type.